rust/datafusion/src/logical_plan/expr.rs

```diff
@@ -271,13 +274,14 @@ impl Expr {
     /// Returns the name of this expression based on [arrow::datatypes::Schema].
     ///
     /// This represents how a column with this expression is named when no alias is chosen
-    pub fn name(&self, input_schema: &Schema) -> Result<String> {
+    pub fn name(&self, input_schema: &DFSchema) -> Result<String> {
         create_name(self, input_schema)
     }
 
     /// Returns a [arrow::datatypes::Field] compatible with this expression.
-    pub fn to_field(&self, input_schema: &Schema) -> Result<Field> {
-        Ok(Field::new(
+    pub fn to_field(&self, input_schema: &DFSchema) -> Result<DFField> {
+        Ok(DFField::new(
+            None, //TODO  qualifier
```

Comment:
might be worth a ticket to track this work -- it would be a good initial project for someone to contribute maybe

Reply:
oops, I actually forgot about that TODO.. thanks